server/CGameHandler.cpp

```diff
@@ -7082,13 +7090,46 @@ void CGameHandler::handleCheatCode(std::string & cheat, PlayerColor player, cons
 		creatures.insert(std::make_pair("vcmiainur", std::make_pair(13, 5))); //5 archangels
 		creatures.insert(std::make_pair("vcmiangband", std::make_pair(66, 10))); //10 black knights
 		creatures.insert(std::make_pair("vcmiglaurung", std::make_pair(133, 5000))); //5000 crystal dragons
+		creatures.insert(std::make_pair("vcmiarchangel", std::make_pair(13, 5))); //5 archangels
+		creatures.insert(std::make_pair("vcmiblackknight", std::make_pair(66, 10))); //10 black knights
+		creatures.insert(std::make_pair("vcmicrystal", std::make_pair(133, 5000))); //5000 crystal dragons
+		creatures.insert(std::make_pair("vcmiazure", std::make_pair(132, 5000))); //5000 azure dragons
+		creatures.insert(std::make_pair("vcmifaerie", std::make_pair(134, 5000))); //5000 faerie dragons
 
 		const CCreature * creature = VLC->creh->objects.at(creatures[cheat].first);
 		for (int i = 0; i < GameConstants::ARMY_SIZE; i++)
 			if (!hero->hasStackAtSlot(SlotID(i)))
 				insertNewStack(StackLocation(hero, SlotID(i)), creature, creatures[cheat].second);
 	}
-	else if (cheat == "vcminoldor")
+	else if (boost::starts_with(cheat, "vcmiarmy"))
+	{
+		cheated = true;
+		if (!hero) return;
+
+		std::vector<std::string> words;
+		boost::split(words, cheat, boost::is_any_of(" "));
+
+		if(words.size() < 2)
+			return;
+
+		std::string creatureIdentifier = words[1];
+
+		auto creature = vstd::tryFindIf(VLC->creh->objects,
+										[creatureIdentifier](auto x)
```

Comment:
Did we agree to use these weird indentations? I still against them. Just waste of space.